test/helpers/setup.go

```diff
@@ -31,7 +31,9 @@ func SetupTestEnv(t *testing.T, ctx context.Context, env environments.Environmen
 		t.Helper()
 
 		t.Log("performing test cleanup")
-		ctx, cancel := context.WithTimeout(context.Background(), time.Minute*5)
+		// Use a longer timeout for cleanup to avoid flakiness when namespaces
+		// take longer to terminate due to finalizers on CRDs (e.g., Konnect resources).
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
```

Comment:
I don't think this will help here, 5 minutes should already be enough to clean up the tests, konnect resources and finalizers.

IMHO if something is taking more, then the issue somewhere else.